core/src/main/java/org/apache/iceberg/rest/RESTMetricsReporter.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.metrics.CommitReport;
+import org.apache.iceberg.metrics.MetricsReport;
+import org.apache.iceberg.metrics.MetricsReporter;
+import org.apache.iceberg.metrics.ScanReport;
+import org.apache.iceberg.relocated.com.google.common.base.Splitter;
+import org.apache.iceberg.rest.requests.ReportMetricsRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class RESTMetricsReporter implements MetricsReporter {
+  private static final Logger LOG = LoggerFactory.getLogger(RESTMetricsReporter.class);
+  private static final Splitter DOT = Splitter.on('.');
+
+  private final RESTClient client;
+  private final ResourcePaths paths;
+  private final Map<String, String> headers;
+
+  RESTMetricsReporter(RESTClient client, ResourcePaths paths, Map<String, String> headers) {
```

Comment:
Won't the headers change when the auth session is refreshed?

Reply:
ah yes, this should be a `Supplier<Map<String, String>>`